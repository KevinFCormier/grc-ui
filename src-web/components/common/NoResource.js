/*******************************************************************************
 * Licensed Materials - Property of IBM
 * (c) Copyright IBM Corporation 2017. All Rights Reserved.
 *
 * Note to U.S. Government Users Restricted Rights:
 * Use, duplication or disclosure restricted by GSA ADP Schedule
 * Contract with IBM Corp.
 *******************************************************************************/
'use strict'

import React from 'react'
import PropTypes from 'prop-types'
import msgs from '../../../nls/platform.properties'
import config from '../../../lib/shared/config'

/* FIXME: Please fix disabled eslint rules when making changes to this file. */
/* eslint-disable react/prop-types */

const NoResource = ({
  title,
  detail,
  children
}, context) =>
  <div className='no-resource'>
    <img className='no-resource-icon' src={`${config.contextPath}/graphics/bee.svg`} alt={msgs.get('svg.description.resource', context.locale)} />
    <div className='no-resource-title'>{title}</div>
    {detail && <div className='no-resource-detail'>{detail}</div>}
    {children}
  </div>

NoResource.contextTypes = {
  locale: PropTypes.string
}

export default NoResource
