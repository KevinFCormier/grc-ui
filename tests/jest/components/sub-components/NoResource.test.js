/*******************************************************************************
 * Licensed Materials - Property of IBM
 * (c) Copyright IBM Corporation 2018, 2019. All Rights Reserved.
 *
 * Note to U.S. Government Users Restricted Rights:
 * Use, duplication or disclosure restricted by GSA ADP Schedule
 * Contract with IBM Corp.
 *******************************************************************************/
'use strict'

import React from 'react'
import NoResource from '../../../../src-web/components/common/NoResource'
import renderer from 'react-test-renderer'

// simple snapshot testing
// This module will create a snapshot in _snapshots_ folder with a file named NoResource.test.js.snap
// https://facebook.github.io/jest/docs/en/snapshot-testing.html
// All snapshot files should be committed alongside the modules they are covering and their tests.
// They should be considered as part of a test, similar to the value of any other assertion in Jest.
// execute `jest --updateSnapshot` to update/re-generate snapshots

describe('NoResource component', () => {
  it('renders as expected', () => {
    const component = renderer.create(
      <NoResource title='title' detail='detailed description'>
        <div className='child'>Test</div>
      </NoResource>
    )
    expect(component.toJSON()).toMatchSnapshot()
  })
})
