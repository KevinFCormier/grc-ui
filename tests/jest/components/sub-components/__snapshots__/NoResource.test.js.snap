// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NoResource component renders as expected 1`] = `
<div
  className="no-resource"
>
  <img
    alt="No Resource"
    className="no-resource-icon"
    src="undefined/common/graphics/no-policy.svg"
  />
  <div
    className="no-resource-title"
  >
    title
  </div>
  <div
    className="no-resource-detail"
  >
    detailed description
  </div>
  <div
    className="child"
  >
    Test
  </div>
</div>
`;
